### ISSUE TITLE
Always route using the id and ignore the slug

### ISSUE THREAD

**Issue by KLuuKer** [VenusInterns/BlogTemplate#127]
Always use the id for routing and getting the post.
When the slug is missing or incorrect just return a **301 Moved Permanently** with the correct\current slug.

As usual scott has a blog post why you should. (he admits his own blog has bad url's in the end)
https://www.hanselman.com/blog/URLsAreUI.aspx

**Comment by jimmylewis** [comment]
Agree on solely depending on the ID for routing.

Could you explain why the 301 is advantageous?  It seems like it would necessitate an extra request for the new page when we already have enough info to resolve the request based on ID.  Why not accept (and ignore) the incorrect slug in the URL?

**Comment by KLuuKer** [comment]
**TL;DR** The only reason would be SEO that's it.

The 301 will make search crawlers that find a old\incorrect url just index it as if it were a direct link to the new\correct url, and when clicking on a search result people will be directly (no redirects) send to the new url.
But more importantly any ranking that you may have build up on the old url will be transferred to the new one.

If you have some (reverse) proxies that are closer to the end user could also potentially cache the redirect and make the extra request return faster.
